core/src/main/java/io/grpc/internal/CallCredentialsApplyingTransportFactory.java

```diff
@@ -78,6 +82,9 @@ protected ConnectionClientTransport delegate() {
     public ClientStream newStream(
         final MethodDescriptor<?, ?> method, Metadata headers, final CallOptions callOptions) {
       CallCredentials creds = callOptions.getCredentials();
+      if (channelCallCredentials != null) {
+        creds = new CompositeCallCredentials(channelCallCredentials, creds);
```

Comment:
`CompositeCallCredentials` checks for both creds != null and `CallOptions.getCredentials` is marked Nullable. Shouldn't you check both for non-null? May be a test should have caught it?